applications/tari_base_node/src/parser.rs

```diff
@@ -1140,14 +1140,25 @@ impl Parser {
             let mut missing_blocks = Vec::new();
             let mut missing_headers = Vec::new();
             print!("Searching for height: ");
+            // We need to check every header, but not every block.
+            let pruning_horizon = height - meta.pruning_horizon;
```

Comment:
I believe pruning horizon has a different definition and horizon height or horizon block should rather be used.
```suggestion
            let horizon_height = meta.horizon_block(height);
```